yoko-core/src/main/java/org/apache/yoko/orb/CORBA/InputStream.java

```diff
@@ -675,25 +683,24 @@ public int available() {
 
     public int read() {
         checkChunk();
-        if (readBuffer.available() < 1) return -1;
-
-        return (0xff & readBuffer.readByte());
+        return readBuffer.available() < 1 ? -1 : readBuffer.readByteAsChar();
     }
 
     public org.omg.CORBA.ORB orb() {
-        if (orbInstance_ != null)
-            return orbInstance_.getORB();
-        return null;
+        return orbInstance_ == null ? null : orbInstance_.getORB();
     }
 
     public boolean read_boolean() {
         checkChunk();
 
-        if (readBuffer.available() < 1) throw newMarshalError(MinorReadBooleanOverflow);
-
-        if (logger.isLoggable(Level.FINEST))
-            logger.finest(String.format("Boolean value is %b from position 0x%x", toBoolean(readBuffer.peekByte()), readBuffer.getPosition()));
-        return toBoolean(readBuffer.readByte());
+        try {
+            int pos = readBuffer.getPosition();
+            byte b = readBuffer.readByte();
+            if (logger.isLoggable(Level.FINEST)) logger.finest(String.format("Boolean value is 0x%08x from position 0x%x", b, pos));
+            return toBoolean(b);
+        } catch (IndexOutOfBoundsException e) {
```

Comment:
Should the IndexOutOfBoundsException be causal of the MARSHAL, or else info from it captured?